### PR TITLE
[MoM] Photokinesis path

### PR DESCRIPTION
**Pull request by Daemonkai**
#### Summary
Mods "Add a new power class to the mind over matter mod"

#### Purpose of change

I made a new class for mind over matter, its theme is photokinesis and it adds 9 powers(spells), 1 passive and the ability for the photokinetic to irradiate food through recipes. 

#### Describe the solution

Powers: 

Snuff light: interrupt the flow of photons to create an area of darkness (darkness field).
Trick of the light: Manipulate the light around you, creating small illusions around your body to become harder to hit.
Photon beam: Concentrate photons into a high-energy beam.
Lucid shadows: Form images of yourself to confuse and distract enemies.
Refraction of arms: Create the illusion of extra arms and wielded items to trick your enemies( +to hit)
Light barrier: Create a barrier of light around yourself to protect against radiation and light damage.(toggle)
Veil of light: Refract light around yourself to become invisible.
Star flash: Create a destructive flash of light. 
Luminous Disintegration: Concentrate a massive amount of light into a single target, causing it to disintegrate.

Passive powers:

Photon regulation: Your powers adapt the amount of photons your eyes receive. (flash/glare/blind protection).
The player can also use his power to irradiate food through the recipe menu, it works for all current irradiated food and requires metaphysics 6 to emulate the mastery over his powers that's required to do this.

Added all the support required like the matrix crystal items, hobbies, professions, feral/civilian psykers etc.

As far as balance is concerned, Standing Storm probably needs to help with a final balance pass if its needed.

#### Testing

Tested every spell and made sure the photokinetic ferals work correctly, crystal and portal awakenings work and the dialogue with the civilian photokinetic functions as it should. 

#### Additional context

It's my first PR and introduction to modding so apologies if I'm doing it wrong.

### CODE REVIEW

File: data/mods/MindOverMatter/monsters/feral_psychics.json
Comment (MNG-cataclysm) [inline data/mods/MindOverMatter/monsters/feral_psychics.json:336]:
```suggestion
    "name": "feral radiant",
```
Since the feral is using photokinesis, I think "radiant" fits a bit more, since it's associated with light.

File: data/mods/MindOverMatter/monsters/feral_psychics.json
Comment (MNG-cataclysm) [inline data/mods/MindOverMatter/monsters/feral_psychics.json:395]:
```suggestion
      "PRIORITIZE_TARGETS"
```
I'd say damage immunity is better saved for the higher tier psykers.

File: data/mods/MindOverMatter/monsters/feral_psychics.json
Comment (MNG-cataclysm) [inline data/mods/MindOverMatter/monsters/feral_psychics.json:475]:
```suggestion
    "name": "feral pulsar",
```
Fits better with the way monsters are named, only a few are capitalized because they're unique or extremely rare.

File: data/mods/MindOverMatter/monsters/monsters_spells.json
Comment (MNG-cataclysm) [inline data/mods/MindOverMatter/monsters/monsters_spells.json:63]:
```suggestion
    "name": "[Ψ]Trick of the Light Monster",
```

File: data/mods/MindOverMatter/mutations/traits.json
Comment (MNG-cataclysm) [inline data/mods/MindOverMatter/mutations/traits.json:29]:
```suggestion
    "description": "You have the power to control the light around you with your mind.",
```

File: data/mods/MindOverMatter/powers/photokinesis.json
Comment (MNG-cataclysm) [inline data/mods/MindOverMatter/powers/photokinesis.json:6]:
```suggestion
    "description": "Interrupt the flow of photons to create an area of darkness",
```

File: data/mods/MindOverMatter/powers/photokinesis.json
Comment (MNG-cataclysm) [inline data/mods/MindOverMatter/powers/photokinesis.json:44]:
```suggestion
    "description": "Manipulate the light around your body, creating small illusions and becoming harder to hit.",
```

File: data/mods/MindOverMatter/powers/photokinesis.json
Comment (MNG-cataclysm) [inline data/mods/MindOverMatter/powers/photokinesis.json:248]:
```suggestion
    "description": "Create a destructive flash of light.",
```

**Review by MNG-cataclysm — CHANGES_REQUESTED** [review]

**Comment by MNG-cataclysm** [comment]
Also, when you're finished editing a JSON file, you need to run the contents through [the linter, found here,](http://[dev.narc.ro](http://dev.narc.ro/cataclysm/format.html)/cataclysm/format.html) so that the style test doesn't return an error.

**Comment by Daemonkai** [comment]
> Also, when you're finished editing a JSON file, you need to run the contents through [the linter, found here,](http://%5Bdev.narc.ro%5D(http://dev.narc.ro/cataclysm/format.html)/cataclysm/format.html) so that the style test doesn't return an error.

So true, my bad.

File: data/mods/MindOverMatter/monsters/feral_psychics.json
Comment (Daemonkai) [reply data/mods/MindOverMatter/monsters/feral_psychics.json]: Supposed to be on par with the other tier3 monsters of the mod which have unique names, so i renamed it to Atom Splitter.

**Comment by Maleclypse** [comment]
@Standing-Storm I'm not sure why reviewbot missed you but adding you in here.

**Comment by Daemonkai** [comment]
> @Standing-Storm I'm not sure why reviewbot missed you but adding you in here.

https://github.com/CleverRaven/Cataclysm-DDA/pull/69092#pullrequestreview-1712935627
I think it didn't miss Storm? Dunno how this work lol

**Comment by Maleclypse** [comment]
> > @Standing-Storm I'm not sure why reviewbot missed you but adding you in here.
> 
> [#69092 (review)](https://github.com/CleverRaven/Cataclysm-DDA/pull/69092#pullrequestreview-1712935627) I think it didn't miss Storm? Dunno how this works lol

There is a tab on the top right that shows reviewers, when I pinged him, Standing-Storm was not on the list I was viewing so I went ahead and pinged him in case. :)

**Review by MNG-cataclysm — APPROVED** [review]

File: data/mods/MindOverMatter/powers/photokinesis.json
Comment (TheShadowFerret) [inline data/mods/MindOverMatter/powers/photokinesis.json:6]:
```suggestion
    "description": "Interrupt the flow of photons to create an area of darkness.",
```

File: data/mods/MindOverMatter/powers/photokinesis.json
Comment (TheShadowFerret) [inline data/mods/MindOverMatter/powers/photokinesis.json:152]:
```suggestion
    "description": "Create the illusion of extra arms and wielded items to trick your enemies.",
```

File: data/mods/MindOverMatter/powers/photokinesis.json
Comment (TheShadowFerret) [inline data/mods/MindOverMatter/powers/photokinesis.json:297]:
```suggestion
    "description": "Concentrate a massive amount of light into a single target, causing it to disintegrate.",
```

File: data/mods/MindOverMatter/professions.json
Comment (TheShadowFerret) [inline data/mods/MindOverMatter/professions.json:67]: Stage magicians should have some social skill 🤔

Comment (Daemonkai) [reply data/mods/MindOverMatter/professions.json:67]: https://github.com/CleverRaven/Cataclysm-DDA/pull/69092/commits/868479fe679aa87778d031fc05ad17e61f74e1a0

**Comment by Standing-Storm** [comment]
Alright, now that Shabbat is over and my daughter is in bed, I've had some time to look at this. As I mentioned on Discord, it's great! It has a unique place and provides new capabilities that other paths don't have (especially around radiation, with a possibility of radio waves, infrared, etc) and I can already think of some stuff I can add to it.

Feedback to come

File: data/mods/MindOverMatter/powers/photokinesis.json
Comment (Standing-Storm) [inline data/mods/MindOverMatter/powers/photokinesis.json:103]:
```suggestion
    "max_level": { "math": [ "int_to_level(1)" ] },
    "field_id": "fd_laser",
    "min_field_intensity": 2,
    "max_field_intensity": 2,
    "field_chance": 1,
    
```
This will leave lasers in the path of the attack for a turn or two.

File: data/mods/MindOverMatter/powers/photokinesis.json
Comment (Standing-Storm) [inline data/mods/MindOverMatter/powers/photokinesis.json:284]:
```suggestion
    "max_level": { "math": [ "int_to_level(1)" ] },
    "field_id": "fd_laser",
    "min_field_intensity": 2,
    "max_field_intensity": 3,
    "field_chance": 1,
```

File: data/mods/MindOverMatter/powers/photokinesis.json
Comment (Standing-Storm) [inline data/mods/MindOverMatter/powers/photokinesis.json:280]:
```suggestion
    "min_aoe":  {
      "math": [ "( (u_val('spell_level', 'spell: photokinetic_light_flash') * 0.3) + 2) * (scaling_factor(u_val('intelligence') ) )" ]
    },
```
Above is just a suggestion, but scaling directly with power level leads to a power that rapidly becomes fatter than it is long based on the range scaling.

File: data/mods/MindOverMatter/powers/photokinesis.json
Comment (Standing-Storm) [inline data/mods/MindOverMatter/powers/photokinesis.json:117]:
```suggestion
    "valid_targets": [ "ground", "hostile", "self", "ally" ],
```

File: data/mods/MindOverMatter/damage_types.json
Comment (Daemonkai) [reply data/mods/MindOverMatter/damage_types.json:9]: Can change it to heat if you prefer it, its a relic of when i was thinking about hardlight too since it could be used in many ways.

File: data/mods/MindOverMatter/monsters/civilian_psychics.json
Comment (Daemonkai) [reply data/mods/MindOverMatter/monsters/civilian_psychics.json:69]: how does "miserable magician" sound to you?

File: data/mods/MindOverMatter/mutations/traits.json
Comment (Daemonkai) [reply data/mods/MindOverMatter/mutations/traits.json:34]: Oh yeah! I was wondering why the starting powers are given at level 2, if you dont mind explaining it to me, will change it regardless to match with the rest.

File: data/mods/MindOverMatter/powers/photokinesis.json
Comment (Daemonkai) [reply data/mods/MindOverMatter/powers/photokinesis.json:220]: Yup great balancing idea, i can see it leading to some stressful situations where you're wondering if it's gonna break at the worst moment and that sounds like very fun gameplay!

File: data/mods/MindOverMatter/powers/photokinesis.json
Comment (Daemonkai) [reply data/mods/MindOverMatter/powers/photokinesis.json:280]: Oh damn, missed that, thanks

**Comment by Daemonkai** [comment]
Moved "fd_darkness" from magiclysm to base game

File: data/mods/MindOverMatter/effects/effects_psionic.json
Comment (Daemonkai) [reply data/mods/MindOverMatter/effects/effects_psionic.json]: check https://github.com/CleverRaven/Cataclysm-DDA/pull/69092/commits/5051940ece560b4beaf623283ad85693b9d39d99
Should be okay now

File: data/mods/MindOverMatter/monsters/civilian_psychics.json
Comment (Standing-Storm) [reply data/mods/MindOverMatter/monsters/civilian_psychics.json:69]: Sounds good!

File: data/mods/MindOverMatter/effects/effects_psionic.json
Comment (Standing-Storm) [reply data/mods/MindOverMatter/effects/effects_psionic.json]: Looks good!

File: data/mods/MindOverMatter/damage_types.json
Comment (Standing-Storm) [reply data/mods/MindOverMatter/damage_types.json:9]: Ohh, now it makes sense. I do think with the current implementation, basing it on heat would be better (maybe like heat 0.75 or something?)

File: data/mods/MindOverMatter/mutations/traits.json
Comment (Standing-Storm) [reply data/mods/MindOverMatter/mutations/traits.json]: I just did that so that the starting powers are a bit more likely to succeed, since (unlike Magiclysm) there's no way to learn later powers without having to use the earlier powers a lot.

File: data/mods/MindOverMatter/powers/photokinesis.json
Comment (Standing-Storm) [reply data/mods/MindOverMatter/powers/photokinesis.json:224]: Alright! I'll roll that into the PR I'm working on and consider this resolved.

**Comment by Daemonkai** [comment]
I've been resolving conversations left and right but i dont actually know the etiquette haha. Do i comment with the changes and wait for approval before resolving? Sorry if i resolved any of your reviews before the proper time folks.

File: data/mods/MindOverMatter/effects/effects_psionic.json
Comment (Daemonkai) [reply data/mods/MindOverMatter/effects/effects_psionic.json]: Oh makes sense, gonna randomize them.

**Comment by Maleclypse** [comment]
> I've been resolving conversations left and right but i dont actually know the etiquette haha. Do i comment with the changes and wait for approval before resolving? Sorry if i resolved any of your reviews before the proper time folks.

Comment with the changes and if I did what the reviewer requested I resolve the conversation.  If I made a change to their request I leave it unresolved for a day or two to give them opportunity to comment on it again.  But that's my personal etiquette I'm not sure we've established a project wide standard.

File: data/mods/MindOverMatter/monsters/civilian_psychics.json
Comment (Daemonkai) [reply data/mods/MindOverMatter/monsters/civilian_psychics.json]: https://github.com/CleverRaven/Cataclysm-DDA/pull/69092/commits/bbbfe7b3f7157996e12b942dca7bde1689ca33ab
done.

File: data/mods/MindOverMatter/mutations/traits.json
Comment (Daemonkai) [reply data/mods/MindOverMatter/mutations/traits.json]: https://github.com/CleverRaven/Cataclysm-DDA/pull/69092/commits/5051940ece560b4beaf623283ad85693b9d39d99
fixed here.

File: data/mods/MindOverMatter/powers/photokinesis.json
Comment (Daemonkai) [reply data/mods/MindOverMatter/powers/photokinesis.json:15]: https://github.com/CleverRaven/Cataclysm-DDA/pull/69092/commits/5051940ece560b4beaf623283ad85693b9d39d99
hopefully the move gets approved.

File: data/mods/MindOverMatter/effects/effects_psionic.json
Comment (Daemonkai) [reply data/mods/MindOverMatter/effects/effects_psionic.json]: https://github.com/CleverRaven/Cataclysm-DDA/pull/69092/commits/7cf6e6bd39181ee53dffe95c89bd60fe97e9d073
should be good now i think.

File: data/mods/MindOverMatter/effects/effects_psionic.json
Comment (Daemonkai) [reply data/mods/MindOverMatter/effects/effects_psionic.json]: https://github.com/CleverRaven/Cataclysm-DDA/pull/69092/commits/7cf6e6bd39181ee53dffe95c89bd60fe97e9d073
done here too.

File: data/mods/MindOverMatter/effects/effects_psionic.json
Comment (Daemonkai) [reply data/mods/MindOverMatter/effects/effects_psionic.json]: https://github.com/CleverRaven/Cataclysm-DDA/pull/69092/commits/5051940ece560b4beaf623283ad85693b9d39d99
3 days sounds good to me.

File: data/mods/MindOverMatter/damage_types.json
Comment (Daemonkai) [reply data/mods/MindOverMatter/damage_types.json]: https://github.com/CleverRaven/Cataclysm-DDA/pull/69092/commits/4c6bc59d83e136efb84699cb2dc3b73ebcd7b8c6
did heat 0.8, there's a chance i'll need to tune up the star flash and disintegration damage a bit.

Comment (Standing-Storm) [reply data/mods/MindOverMatter/damage_types.json]: That was the reason I suggested a lower rate, though when I checked for pyrokinesis, heat armor is pretty rare among monsters so it's not likely to lead to any odd issues by being based on heat. 

Star Flash damage did seem low when I tested it--maybe higher per-level scaling on that and Disintegrate would help?

Comment (Daemonkai) [reply data/mods/MindOverMatter/damage_types.json]: https://github.com/CleverRaven/Cataclysm-DDA/pull/69092/commits/e77ad3f106d56d4b2c4612033dd28b4d729d14b3
Damage numbers are a bit higher now.

Comment (Standing-Storm) [reply data/mods/MindOverMatter/damage_types.json]: Looks good!

**Review by Standing-Storm — APPROVED** [review]

**Pull request merged**
